src/file-upload/file-list.tsx

```diff
@@ -97,26 +107,30 @@ export const FileList = ({
     // =========================================================================
     // EVENT HANDLERS
     // =========================================================================
-    const handleDescriptionUpdate =
-        (item: FileItemProps) => (description: string) => {
-            const updatedItem = { ...item, description };
-            onItemUpdate(updatedItem);
-        };
+    const handleSaveEdit = (item: FileItemProps) => (description: string) => {
+        updateRenderModes(item.id, "display");
+        removeDescription(item.id);
 
-    const handleCancel = (item: FileItemProps) => () => {
-        const updatedRenderModes = { ...renderModes };
-        updatedRenderModes[item.id] = item.description
-            ? "list"
-            : "cancelled-edit";
+        const updatedItem = { ...item, description };
+        onItemUpdate(updatedItem);
+    };
 
-        setRenderModes(updatedRenderModes);
+    const handleBlurEdit = (item: FileItemProps) => (value: string) => {
+        descriptionsValueRef.current[item.id] = value;
     };
 
-    const handleInitiateEdit = (item: FileItemProps) => () => {
-        const updatedRenderModes = { ...renderModes };
-        updatedRenderModes[item.id] = "edit";
+    const handleCancel = (item: FileItemProps) => () => {
+        if (!item.description || item.description.length === 0) {
+            // New addition
+            onItemDelete(item);
```

Comment:
`removeDescription` should also be run here, as the item id might have been set previously

Reply:
It will be unlikely so but i think I should do so given it could be altered by the parent.